storage/storage-interface/src/lib.rs

```diff
@@ -664,6 +665,7 @@ pub trait DbWriter: Send + Sync {
         base_state_version: Option<Version>,
         ledger_info_with_sigs: Option<&LedgerInfoWithSignatures>,
         save_state_snapshots: bool,
+        checkpoint: SparseMerkleTree<StateValue>,
```

Comment:
nit: naming: "state_checkpoint" / "state_tree"?

preferring "state_tree" (like what you used in `ExecutedTrees::state_tree()`, since the tree at the end of a state sync chunk does not map to a state checkpoint txn..